### PR TITLE
Project details page

### PR DESCRIPTION
**Pull request by narisettiashok**
New Project Details page updated

### CODE REVIEW

**Comment by narisettiashok** [comment]
There is a problem in submitting the form

File: src/components/Form/ProjectsDataDatePicker.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import DateView from 'react-datepicker';
```
Comment (deepakrajendra) [inline src/components/Form/ProjectsDataDatePicker.jsx]: DatePicker is a better name than DateView

File: src/components/Form/ProjectsDataDatePicker.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                <Field name={name}>
+                    {
+                        ({ form, field }) => {
+                            const { setFieldValue } = form
+                            const { value } = field
+                            return (
+                                <DateView
+                                    className="focus:outline-none"
+                                    id={name} {...field} {...rest}
+                                    selected={value}
```
Comment (deepakrajendra) [inline src/components/Form/ProjectsDataDatePicker.jsx]: Change to selected={(value && new Date(value)) || null}

File: src/components/Form/TextError.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React from 'react'
```
Comment (deepakrajendra) [inline src/components/Form/TextError.jsx]: Change to double quotes

File: src/components/Projects/BOQProject.jsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import FormControl from "../Form/FormControl";
+import LoadingSpinner from "../LoadingSpinner";
+
+
+function BOQProject(props) {
+
+    const formikInitialValues = {
+        projectName: "",
+        projectCost: "",
+        ProjectAgreementDate: "",
```
Comment (deepakrajendra) [inline src/components/Projects/BOQProject.jsx]: ALWAYS use camelCase except for component name

File: src/components/Projects/EPCProject.jsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                                        type="text"
+                                        label="Project Cost"
+                                        id="projectCost"
+                                        name="projectCost"
+                                        placeholder="Enter Project Cost in 'Rupees'"
+                                    />
+                                    <FormControl
+                                        control="date"
+                                        label="Project Agreement Date"
+                                        name="projectAgreementDate"
```
Comment (deepakrajendra) [inline src/components/Projects/EPCProject.jsx]: Doesn't match with formik field name. Always use camelcase

File: src/components/Projects/EPCProject.jsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                                        label="Project Completion Period"
+                                        id="ProjectCompletionPeriod"
+                                        name="ProjectCompletionPeriod"
+                                        placeholder="Project Completion Period 'Months'"
+                                    />
+
+                                    <div className="text-center mt-6">
+                                        <button className="bg-blue-800 text-white active:bg-blue-600 text-sm font-bold uppercase px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 w-full ease-linear transition-all duration-150"
+                                            type="submit" disabled={!formik.isValid || formik.isSubmitting}>
+                                            Submit Data Here {formik.isSubmitting && <LoadingSpinner />}
```
Comment (deepakrajendra) [inline src/components/Projects/EPCProject.jsx]: Change text to 'Done'

**Comment by deepakrajendra** [comment]
[Add Project UI](https://trello.com/c/k2q1q05s/16-add-project-ui)

File: src/components/Form/ProjectsDataDatePicker.jsx (hunk 1 above)
Comment (narisettiashok) [reply src/components/Form/ProjectsDataDatePicker.jsx]: Resolved

File: src/components/Form/ProjectsDataDatePicker.jsx (hunk 2 above)
Comment (narisettiashok) [reply src/components/Form/ProjectsDataDatePicker.jsx]: Resolved

File: package.json [hunk 7]
```diff
@@ -9,9 +9,12 @@
     "@testing-library/jest-dom": "^5.12.0",
     "@testing-library/react": "^11.2.6",
     "@testing-library/user-event": "^12.8.3",
+    "ag-grid-community": "^25.3.0",
```
Comment (deepakrajendra) [inline package.json]: Remove table-related libraries. Add them when tables are used

File: src/components/Form/FormControl.jsx [hunk 8]
```diff
@@ -1,14 +1,20 @@
 import React from "react";
 import Checkbox from "./Checkbox";
 import Input from "./Input";
+import ProjectsDataDatePicker from "./ProjectsDataDatePicker";
+import ProjectsDataInput from './ProjectsDataInput';
```
Comment (deepakrajendra) [inline src/components/Form/FormControl.jsx]: Use double quotes everywhere. Keep it consistent

File: src/components/Form/ProjectsDataDatePicker.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
```
Comment (deepakrajendra) [inline src/components/Form/ProjectsDataDatePicker.jsx]: You dont have to import react in every file

File: src/components/Form/FormControl.jsx (hunk 8 above)
Comment (deepakrajendra) [reply src/components/Form/FormControl.jsx]: Change in all files

File: src/components/Projects/BOQProject.jsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
+import { Form, Formik } from "formik";
+import * as Yup from "yup";
+
+import FormControl from "../Form/FormControl";
+import LoadingSpinner from "../LoadingSpinner";
+
```
Comment (deepakrajendra) [inline src/components/Projects/BOQProject.jsx]: Remove this line

File: src/components/Projects/BOQProject.jsx [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+    const onSubmit = (values, onSubmitProps) => {
+        setTimeout(() => {
+            onSubmitProps.setSubmitting(false);
+            onSubmitProps.resetForm();
+            props.history.push("/app/projects");
+        }, 1000);
+        console.log('Form Data', values);
+    };
+
+
```
Comment (deepakrajendra) [inline src/components/Projects/BOQProject.jsx]: Remove this line

File: src/components/Projects/NewProject.jsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+                    hover:bg-blue-900 hover:text-white" to="/app/epc">
+                    <span>EPC</span>
+                </Link>
+                <Link className="rounded-lg p-3 flex-1 items-center space-x-3 bg-blue-700 text-white cursor-pointer m-2
+                    hover:bg-blue-900 hover:text-white" to="/app/boq">
+                    <span>BOQ</span>
+                </Link>
+            </div>
+        </div>
+
```
Comment (deepakrajendra) [inline src/components/Projects/NewProject.jsx]: Remove this line

File: src/components/Projects/ProjectsPage.jsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+                    hover:bg-blue-900 hover:text-white" to='/app/newProject'>
+                    <span>New Project</span>
+                </Link>
+                <Link className="rounded-lg p-3 flex-1 items-center space-x-3 bg-blue-700 text-white cursor-pointer m-2
+                    hover:bg-blue-900 hover:text-white" to='/app/existingProject'>
+                    <span>Existing Project</span>
+                </Link>
+            </div>
+        </div>
+
```
Comment (deepakrajendra) [inline src/components/Projects/ProjectsPage.jsx]: Remove this line

**Pull request merged**